diff --git a/src/thread/pthread/SDL_systhread.c b/src/thread/pthread/SDL_systhread.c
index 3b3f40f..6d326b8 100644
--- a/src/thread/pthread/SDL_systhread.c
+++ b/src/thread/pthread/SDL_systhread.c
@@ -148,10 +148,10 @@ SDL_SYS_SetupThread(const char *name)
             #if defined(__NETBSD__)
             pthread_setname_np(pthread_self(), "%s", name);
             #else
-            if (pthread_setname_np(pthread_self(), name) == ERANGE) {
+            if (pthread_setname_np(name) == ERANGE) {
                 char namebuf[16]; /* Limited to 16 char */
                 SDL_strlcpy(namebuf, name, sizeof (namebuf));
-                pthread_setname_np(pthread_self(), namebuf);
+                pthread_setname_np(namebuf);
             }
             #endif
         #elif HAVE_PTHREAD_SET_NAME_NP
diff --git a/src/video/cocoa/SDL_cocoawindow.m b/src/video/cocoa/SDL_cocoawindow.m
index e9e3c09..5532561 100644
--- a/src/video/cocoa/SDL_cocoawindow.m
+++ b/src/video/cocoa/SDL_cocoawindow.m
@@ -1779,6 +1779,9 @@ - (BOOL)acceptsFirstMouse:(NSEvent *)theEvent
     }
 #endif /* SDL_VIDEO_OPENGL_EGL */
 #endif /* SDL_VIDEO_OPENGL_ES2 */
+    // force layer
+    [contentView setWantsLayer:TRUE];
+    
     [nswindow setContentView:contentView];
 
     if (SetupWindowData(_this, window, nswindow, contentView, SDL_TRUE) < 0) {
